File: source/dotnet/Library/AdaptiveCards/AdaptiveCardParseResult.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
using System.Collections.Generic;

namespace AdaptiveCards
{
    /// <summary>
    /// The result of parsing an AdaptiveCard from JSON.
    /// </summary>
    public class AdaptiveCardParseResult
    {
        /// <summary>
        /// The parsed card object.
        /// </summary>
        public AdaptiveCard Card { get; set; }

        /// <summary>
        /// A collection of any warnings that might have been encountered when parsing the card.
        /// </summary>
        public List<AdaptiveWarning> Warnings { get; } = new List<AdaptiveWarning>();
    }
}

File: source/uwp/AdaptiveCardsObjectModel/lib/AdaptiveCardParseResult.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
#include "pch.h"
#include "AdaptiveCardParseResult.h"
#include "AdaptiveCardParseResult.g.cpp"

namespace winrt::AdaptiveCards::ObjectModel::Uwp::implementation
{
    AdaptiveCardParseResult::AdaptiveCardParseResult(winrt::AdaptiveCards::ObjectModel::Uwp::AdaptiveCard const& card) :
        AdaptiveCard{card}, Errors{winrt::single_threaded_vector<Uwp::AdaptiveError>()}, Warnings{winrt::single_threaded_vector <Uwp::AdaptiveWarning>()}
    {
    }
}
